dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/icon/IconController.java

```diff
@@ -124,17 +130,15 @@ public void getIconData(@PathVariable String key, HttpServletResponse response)
   }
 
   @GetMapping
-  public @ResponseBody List<IconResponse> getAllIcons(
-      @RequestParam(required = false) String[] keywords) {
-    List<Icon> icons;
-
-    if (keywords == null) {
-      icons = iconService.getIcons();
-    } else {
-      icons = iconService.getIcons(keywords);
-    }
+  public @ResponseBody List<ObjectNode> getAllIcons(IconRequestParams iconRequestParams)
```

Comment:
github seems to have lost our conversation on this one. https://github.com/dhis2/dhis2-core/pull/15929#discussion_r1450022903 still applies